### PR TITLE
[Enhancement] Fix some type conversion warning

### LINKED ISSUE #12293: [Enhancement] Fix some type conversion warning
## Enhancement
Fix some type conversion warning, to avoid some hidden problems maybe behind them.

### PR DESCRIPTION
**Pull request by luohaha**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
Fixes #12293

## Problem Summary(Required) ：
Fix some type conversion warning, to avoid some hidden problems maybe behind them.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by luohaha** [comment]
run starrocks_clang-tidy

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: be/src/common/logconfig.cpp [hunk 1]
```diff
@@ -38,11 +38,11 @@ static bool logging_initialized = false;
 static std::mutex logging_mutex;
 
 static bool iequals(const std::string& a, const std::string& b) {
-    unsigned int sz = a.size();
+    std::size_t sz = a.size();
```
Comment (trueeyu) [inline be/src/common/logconfig.cpp:41]: Noe need add std?  std::size_t --> size_t

File: be/src/common/logconfig.cpp [hunk 2]
```diff
@@ -206,11 +206,11 @@ void shutdown_logging() {
 
 std::string FormatTimestampForLog(MicrosecondsInt64 micros_since_epoch) {
     time_t secs_since_epoch = micros_since_epoch / 1000000;
-    int usecs = micros_since_epoch % 1000000;
+    long int usecs = micros_since_epoch % 1000000;
```
Comment (trueeyu) [inline be/src/common/logconfig.cpp:209]: int64_t?

Comment (trueeyu) [reply be/src/common/logconfig.cpp:209]: typedef int64 MicrosecondsInt64;

File: be/src/storage/key_coder.h [hunk 3]
```diff
@@ -327,7 +327,7 @@ class KeyCoderTraits<OLAP_FIELD_TYPE_CHAR> {
     static void encode_ascending_datum(const Datum& value, size_t index_size, std::string* buf) {
         const Slice& slice = value.get_slice();
         buf->append(slice.data, slice.size);
-        int pad = index_size > slice.size ? index_size - slice.size : 0;
+        std::size_t pad = index_size > slice.size ? index_size - slice.size : 0;
```
Comment (trueeyu) [inline be/src/storage/key_coder.h:330]: remove std?

Comment (luohaha) [reply be/src/storage/key_coder.h]: Done.

File: be/src/common/logconfig.cpp (hunk 1 above)
Comment (luohaha) [reply be/src/common/logconfig.cpp]: Done

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

File: be/src/common/daemon.cpp [hunk 4]
```diff
@@ -308,8 +308,8 @@ void Daemon::init(int argc, char** argv, const std::vector<StorePath>& paths) {
 
 void Daemon::stop() {
     _stopped.store(true, std::memory_order_release);
-    int thread_size = _daemon_threads.size();
-    for (int i = 0; i < thread_size; ++i) {
+    std::size_t thread_size = _daemon_threads.size();
```
Comment (chaoyli) [inline be/src/common/daemon.cpp:311]:
```suggestion
    size_t is enough
```

File: be/src/common/daemon.cpp [hunk 5]
```diff
@@ -308,8 +308,8 @@ void Daemon::init(int argc, char** argv, const std::vector<StorePath>& paths) {
...
+    for (std::size_t i = 0; i < thread_size; ++i) {
```
Comment (chaoyli) [inline be/src/common/daemon.cpp:312]:
```suggestion
    for (size_t i = 0; i < thread_size; ++i) {
```

File: be/src/common/daemon.cpp (hunk 4 above)
Comment (luohaha) [reply be/src/common/daemon.cpp]: Done

**Review by trueeyu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
starrocks_be_unittest succeeded.

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**